website/src/models/group.js

```diff
@@ -404,11 +404,29 @@ GroupSchema.methods.leave = function(user, keep, mainCb){
           group.type === 'party' ||
           (group.type === 'guild' && group.privacy === 'private')
       )){
-        // TODO remove invitations to this group
-        groupWasRemoved = true;
-        Group.remove({
-          _id: group._id
-        }, cb);
+        async.waterfall([
```

Comment:
I think in this case we can do it here ad it's the only place ( so far ) where a group is deleted. But for future changes maybe it's better a mideware